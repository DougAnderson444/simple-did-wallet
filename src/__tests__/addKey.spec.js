import didWallet from '../index.js'

import fixtures from './__fixtures__'

describe('addKey', () => {
  it('can addKey to empty wallet', () => {
    const wallet = didWallet.create()
    expect(wallet.keys).toEqual({})

    wallet.addKey({
      type: 'assymetric',
      encoding: 'hex',
      publicKey: fixtures.secp256k1_keypair_0.publicKey,
      privateKey: fixtures.secp256k1_keypair_0.privateKey,
      tags: ['Secp256k1VerificationKey2018', 'did:example:456', 'A'],
      notes: ''
    })

    expect(wallet.keys).toMatchSnapshot()
  })

  it('can addKey to existing wallet', () => {
    const wallet = didWallet.create({
      keys: [
        {
          type: 'assymetric',
          encoding: 'hex',
          publicKey: fixtures.secp256k1_keypair_0.publicKey,
          privateKey: fixtures.secp256k1_keypair_0.privateKey,
          tags: ['Secp256k1VerificationKey2018', 'did:example:456', 'A'],
          notes: ''
        }
      ]
    })

    expect(wallet.keys).toMatchSnapshot()

    wallet.addKey({
      type: 'assymetric',
      encoding: 'application/pgp-keys',
      publicKey: fixtures.openpgpg_keypair_1.publicKey,
      privateKey: fixtures.openpgpg_keypair_1.privateKey,
      revocationCertificate: fixtures.openpgpg_keypair_1.revocationCertificate,
      tags: ['Secp256k1VerificationKey2018', 'did:example:456', 'A'],
      notes: ''
    })

    expect(wallet.keys).toMatchSnapshot()
  })
})
