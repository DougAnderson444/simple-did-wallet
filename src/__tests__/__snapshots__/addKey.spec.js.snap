// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addKey can addKey to empty wallet 1`] = `
Object {
  "WgGaIZApxh9aeV9K51GfGEMNEkgS6av5boqFZI0QHHA": Object {
    "encoding": "hex",
    "kid": "WgGaIZApxh9aeV9K51GfGEMNEkgS6av5boqFZI0QHHA",
    "notes": "",
    "privateKey": "68be85c3ca96c4f67056e57c1a6d9169734b54ab87ed9a853e349461da0e8c0e",
    "publicKey": "0286cdfebc43519c64baf08017b9f14a66871ee907770e6bc8fcd62282ebc72dc0",
    "tags": Array [
      "Secp256k1VerificationKey2018",
      "did:example:456",
      "A",
    ],
    "type": "assymetric",
  },
}
`;
